### ISSUE TITLE
render_site issues warning message and doesn't render html properly when called from a working directory other than the site directory

### ISSUE THREAD

**Issue by adamcagle** [rstudio/rmarkdown#1528]
When I call rmarkdown::render_site("SiteDirectory") when my working directory isn't SiteDirectory, I get the following error:

 ```{r}
Output created: _site/index.html
Warning message:
In res[i] <- withCallingHandlers(if (tangle) process_tangle(group) else process_group(group),  :
  number of items to replace is not a multiple of replacement length
```
In addition, my documents don't render correctly. However, if I call render_site() when my working directory is the site directory, there are no issues.

I deleted everything I could to see if I could isolate the problem, and I'm attaching a zip file [example.zip](https://github.com/rstudio/rmarkdown/files/2834540/example.zip) with the results of that exercise. Here's what I noticed in addition to the working directory issue.

* If you delete one of the blank lines in the bash code chunk in "Breaks Things.Rmd", the warning message goes away.
* Or, if you rename the "Breaks Things.Rmd" file to some name without a space in it, the warning message also goes away.

Here is my session info:
```{r}
R version 3.5.2 (2018-12-20)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.1 LTS, RStudio 1.1.463

Locale:
  LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
  LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

Package version:
  base64enc_0.1.3 digest_0.6.18   evaluate_0.12   glue_1.3.0      graphics_3.5.2  grDevices_3.5.2 highr_0.7      
  htmltools_0.3.6 jsonlite_1.6    knitr_1.21      magrittr_1.5    markdown_0.9    methods_3.5.2   mime_0.6       
  Rcpp_1.0.0      rmarkdown_1.11  stats_3.5.2     stringi_1.2.4   stringr_1.3.1   tinytex_0.9     tools_3.5.2    
  utils_3.5.2     xfun_0.4        yaml_2.2.0     

Pandoc version: 1.19.2.1
```

**Comment by yihui** [comment]
Thanks so much! This is the one of the best bug reports I have ever seen. You have done a great job of simplifying it to [the very minimal example](https://yihui.name/en/2017/09/the-minimal-reprex-paradox/)! I really appreciate it.

You can install the development version of **knitr** via

```r
remotes::install_github('yihui/knitr')
```

**Comment by adamcagle** [comment]
Thanks, Yihui! The contributing guide was very helpful. I'm new to this, so the guidance made me feel a lot more comfortable posting an issue. Going through the exercise of making a minimal example, I see now how this is a great way to approach any problem. I was able to rule out all of my theories about what might be causing the issue really quickly this way, and I never would have guessed what the end result would turn out to be. It would have saved me a lot of time had I used this technique to start with!

Thank you for all of your work on rmarkdown! It's a great resource, and I've gotten a lot of mileage out of it.
